fsspec/implementations/reference.py

```diff
@@ -213,27 +295,63 @@ def items(self):
         return RefsItemsView(self)
 
     def __getitem__(self, key):
-        if key in self._items:
-            val = self._items[key]
-            if val is None:
-                raise KeyError
-        if key in self.zmetadata:
-            # spec requires bytes even if we already decoded the metadata
-            return json.dumps(self.zmetadata[key]).encode()
         return self._load_one_key(key)
 
     def __setitem__(self, key, value):
-        self._items[key] = value
+        field, chunk = key.split("/")
+        is_meta = chunk.startswith(".z")
+        if field and not is_meta:
+            record, _, _ = self._key_to_record(key)
+            subdict = self._items.setdefault((field, record), {})
+            subdict[chunk] = value
+            if len(subdict) == self.record_size:
+                self.write(field, record)
+        else:
+            # metadata or top-level
+            self._items[key] = value
 
     def __delitem__(self, key):
+        field, chunk = key.split("/")
+        is_meta = chunk.startswith(".z")
         if key in self._items:
             del self._items[key]
         elif key in self.zmetadata:
             del self.zmetadata[key]
         else:
-            # TODO: Add method to RefFs that tracks changes in _items
-            # and updates / copies appropriate parquet ref files.
-            self._items[key] = None
+            if field and not is_meta:
+                record, _, _ = self._key_to_record(key)
+                subdict = self._items.setdefault((field, record), {})
+                subdict[chunk] = None
+                if len(subdict) == self.record_size:
+                    self.write(field, record)
+            else:
+                # metadata or top-level
+                self._items[key] = None
+
+    def write(self, field, record, base_url=None, storage_options=None):
+        # TODO: if the dict is incomplete, also load records and merge in
```

Comment:
Ah, indeed - if we already have an fs, we don't need storage options at all. I do want to leave open, that we might ead from one place and write to another; but MultiZarrToZarr doesn't actually need that, since it only writes.